src/index.rs

```diff
@@ -231,8 +235,63 @@ impl IndexTable {
 		Ok(try_io!(Ok(&map[offset..offset + CHUNK_LEN])))
 	}
 
-	#[inline(never)]
 	fn find_entry(&self, key_prefix: u64, sub_index: usize, chunk: &[u8]) -> (Entry, usize) {
+		#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+		if is_x86_feature_detected!("sse2") {
+			return self.find_entry_sse2(key_prefix, sub_index, chunk)
+		}
+		self.find_entry_regular(key_prefix, sub_index, chunk)
+	}
+
+	#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+	fn find_entry_sse2(&self, key_prefix: u64, sub_index: usize, chunk: &[u8]) -> (Entry, usize) {
+		debug_assert!(
+			(CHUNK_ENTRIES - sub_index) % 4 == 0,
+			"SSE2 find entry only works on number of entities dividable by 4"
+		);
+
+		unsafe {
+			let partial_key = Entry::extract_key(key_prefix, self.id.index_bits());
+			let target = _mm_set1_epi32(partial_key as i32);
+			let shift = Entry::address_bits(self.id.index_bits());
+			let mut i = sub_index;
+			while i + 4 <= CHUNK_ENTRIES {
+				// We load the value two by two and do the same shift as for the partial_key
+				// before moving the values into the low part of the register
+				let first_two = _mm_shuffle_epi32::<0b11011000>(_mm_srl_epi64(
```

Comment:
I am a bit lost in my reading, I would get 1 and 3 at first then both get interleaves, so would expect 0b0111...
(before shift the partial is pos 0 and 2 and then 1 and 3).
But I might mess my ordering here.